### PR TITLE
Skip UI tests if a required UI API Layer module is disabled

### PR DESCRIPTION
**Pull request by pkosiec**
**Description**

Changes proposed in this pull request:

- Skip UI tests if a required UI API Layer module is disabled

**Related issue(s)**
See https://github.com/kyma-project/kyma/issues/2236

### CODE REVIEW

File: tests/ui-tests/setup/kubeconfig.js [hunk 1]
```diff
@@ -17,3 +17,5 @@ export const loadKubeConfig = () => {
   kubeConfig.loadFromDefault();
   return kubeConfig;
 };
+
+export const kubeConfig = loadKubeConfig();
```
Comment (magicmatatjahu) [inline tests/ui-tests/setup/kubeconfig.js]: Awesome!

File: tests/ui-tests/setup/test-pluggable.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    return this.modules.includes(moduleName);
+  }
+}
+
+const moduleFetcher = new BackendModuleFetcher();
+
+export const testPluggable = (requiredModule, name, fn) => {
+  return test(name, async done => {
```
Comment (magicmatatjahu) [inline tests/ui-tests/setup/test-pluggable.js]:
``` js
return test(name, async done => {
  if (await isModuleEnabled(requiredModule)) return await fn(done);

  console.log(
    `Module ${requiredModule} is disabled. Skipping test ${name}...`
  );
  return done();
}
```

File: tests/ui-tests/tests/catalog-ui-test.js [hunk 3]
```diff
@@ -19,6 +22,13 @@ let token = '';
 
 describeIf(dex.isStaticUser(), 'Catalog basic tests', () => {
   beforeAll(async () => {
+    if (!(await isModuleEnabled(REQUIRED_MODULE))) {
+      console.log(
+        `Skipping 'beforeAll' because of disabled module ${REQUIRED_MODULE}`
```
Comment (magicmatatjahu) [inline tests/ui-tests/tests/catalog-ui-test.js]: Please create function for that with `REQUIRED_MODULE ` parameter in `test-pluggable` file

File: tests/ui-tests/setup/kubeconfig.js (hunk 1 above)
Comment (pkosiec) [reply tests/ui-tests/setup/kubeconfig.js]: ?

File: tests/ui-tests/tests/lambda-ui-test.js [hunk 4]
```diff
@@ -5,28 +5,47 @@ import common from '../commands/common';
...
     const data = await common.beforeAll();
     browser = data.browser;
     page = data.page;
     logOnEvents(page, t => (token = t));
   });
 
   afterAll(async () => {
+    if (!(await isModuleEnabled(REQUIRED_MODULE))) {
+      console.log(
+        `Skipping 'beforeAll' because of disabled module ${REQUIRED_MODULE}`
```
Comment (magicmatatjahu) [inline tests/ui-tests/tests/lambda-ui-test.js]: `Skipping 'afterAll' ...`. Please fix it in other places.

File: tests/ui-tests/setup/test-pluggable.js (hunk 2 above)
Comment (pkosiec) [reply tests/ui-tests/setup/test-pluggable.js]: No. I want to keep happy path on the left side 😉

**Pull request merged**